message/decorator_test.go

```diff
@@ -0,0 +1,128 @@
+package message_test
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ThreeDotsLabs/watermill/message/subscriber"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/ThreeDotsLabs/watermill/message/infrastructure/gochannel"
+)
+
+var noop = func(*message.Message) {}
+
+func TestMessageTransformer_Subscribe(t *testing.T) {
+	numMessages := 1000
+	pubsub := gochannel.NewGoChannel(0, watermill.NewStdLogger(true, true), time.Second)
+
+	onMessage := func(msg *message.Message) {
+		msg.Metadata.Set("key", "value")
+	}
+	decorator := message.MessageTransformSubscriberDecorator(onMessage)
+
+	decoratedSub, err := decorator(pubsub.(message.Subscriber))
+	require.NoError(t, err)
+
+	messages, err := decoratedSub.Subscribe("topic")
+	require.NoError(t, err)
+
+	go func() {
+		for i := 0; i < numMessages; i++ {
+			err = pubsub.Publish("topic", message.NewMessage(strconv.Itoa(i), []byte{}))
+			require.NoError(t, err)
+		}
+	}()
+
+	received, all := subscriber.BulkRead(messages, numMessages, time.Second)
+	require.True(t, all)
```

Comment:
you check count here only, `tests.AssertAllMessagesReceived` is also good idea

Reply:
aaand should probably go to `TestMessageTransformer_transparent`, you may also consider to move `TestMessageTransformer_transparent` at the bottom (IMO it's better to start read with simpler tests)

Reply:
bc88f5f2868b677ed40b50f9eca641889cb6a64d reorganized the tests